lib/msal-browser/src/interaction_client/SilentAuthCodeClient.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AuthenticationResult, ICrypto, Logger, CommonAuthorizationCodeRequest, AuthorizationCodeClient, AuthError } from "@azure/msal-common";
+import { StandardInteractionClient } from "./StandardInteractionClient";
+import { AuthorizationUrlRequest } from "../request/AuthorizationUrlRequest";
+import { BrowserConfiguration } from "../config/Configuration";
+import { BrowserCacheManager } from "../cache/BrowserCacheManager";
+import { EventHandler } from "../event/EventHandler";
+import { INavigationClient } from "../navigation/INavigationClient";
+import { BrowserAuthError } from "../error/BrowserAuthError";
+import { InteractionType, ApiId } from "../utils/BrowserConstants";
+import { SilentHandler } from "../interaction_handler/SilentHandler";
+import { AuthorizationCodeRequest } from "../request/AuthorizationCodeRequest";
+
+export class SilentAuthCodeClient extends StandardInteractionClient {
+    private apiId: ApiId;
+
+    constructor(config: BrowserConfiguration, storageImpl: BrowserCacheManager, browserCrypto: ICrypto, logger: Logger, eventHandler: EventHandler, navigationClient: INavigationClient, apiId: ApiId, correlationId?: string) {
+        super(config, storageImpl, browserCrypto, logger, eventHandler, navigationClient, correlationId);
+        this.apiId = apiId;
+    }
+    
+    /**
+     * Acquires a token silently by opening a hidden iframe to the /authorize endpoint with prompt=none
+     * @param request 
+     */
+    async acquireToken(request: AuthorizationCodeRequest): Promise<AuthenticationResult> {
+        this.logger.trace("SilentAuthCodeClient.acquireToken called");
+
+        // Auth code payload is required
+        if (!request.code) {
+            throw BrowserAuthError.createAuthCodeRequiredError();
```

Comment:
This was thrown earlier as well. Do we need this logic in both places?

Reply:
Did this mostly for completeness. In practice, this shouldn't be hit.